.github/workflows/concrete_python_release.yml

```diff
@@ -198,17 +199,55 @@ jobs:
           path: frontends/concrete-python/dist/*macos*.whl
           retention-days: 3
 
-  push:
+  hash:
+    # Generate hashes for the wheels, used later for provenance.
     needs: [build-linux-x86, build-macos]
     runs-on: ubuntu-latest
+    outputs:
+      hash: ${{ steps.hash.outputs.hash }}
+    steps:
+      - uses: actions/download-artifact@65a9edc5881444af0b9093a5e628f2fe47ea3b2e # v4.1.7
+        with:
+          path: frontends/concrete-python/dist
+          pattern: wheel-*
+          merge-multiple: true
+      - name: generate hash
+        id: hash
+        run: cd frontends/concrete-python/dist && echo "hash=$(sha256sum *.whl | base64 -w0)" >> $GITHUB_OUTPUT
+
+  provenance:
+    needs: [hash]
+    permissions:
+      actions: read
+      id-token: write
+      contents: write
+    # Can't pin with hash due to how this workflow works.
+    uses: slsa-framework/slsa-github-generator/.github/workflows/generator_generic_slsa3.yml@v2.0.0
+    with:
+      base64-subjects: ${{ needs.hash.outputs.hash }}
+
+  push:
+    needs: [build-linux-x86, build-macos, provenance]
+    runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v3
 
       - uses: actions/download-artifact@v4
         with:
           path: wheels
           merge-multiple: true
-
+      - uses: actions/download-artifact@v4
+        with:
+          pattern: '*.intoto.jsonl'
+      # When building a new tag, create a new draft release.
+      - name: create draft release
+        if: ${{ env.RELEASE_TYPE == 'public' || env.RELEASE_TYPE == 'nightly' }}
```

Comment:
@youben11 How can I restrict this step to when it runs on a tag? (I don't see any check for this in the current wf)
Should I test it directly here with a  `contains(github.ref, 'refs/tags/')`?

Reply:
ok I see the tag is pushed from this wf, right now I'm creating a GH release for both types of release, maybe I should restrict to public? (let's keep it like that for test and another PR to move it to public only when ok)